### PR TITLE
Add verification tests for DeriveContext

### PR DESCRIPTION
**Pull request by sree-revoori1**
This PR is just https://github.com/chipsalliance/caliptra-dpe/pull/250 rebased, along with a few small changes. Will follow up with a PR for DeriveContext, recursive = true. 

Co-authored-by: haripriyab@ami.com

### CODE REVIEW

File: verification/testing/deriveContext.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+	handle = &resp.NewContextHandle
+	parentHandle := resp.ParentContextHandle
+
+	if parentHandle == client.RetiredContextHandle {
+		t.Errorf("[ERROR]: Incorrect handle. Should return retained handle, but returned invalidated handle %v", parentHandle)
+	}
+}
+
+// Validates DerivedChild command with ChangeLocality flag.
+func TestChangeLocality(d client.TestDPEInstance, c client.DPEClient, t *testing.T) {
```
Comment (jhand2) [inline verification/testing/deriveContext.go:80]: I think this test is irreversible as-is. To change this we would need to derive a non-default context that we can destroy before making derive calls.

We can open an issue for that and just mark as irreversible for now.

File: verification/testing/deriveContext.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+		}
+		handle = &resp.NewContextHandle
+	} else {
+		t.Skip("[WARNING]: Profile has no support for \"InternalInputInfo\" flag. Skipping the validation for this flag...")
+	}
+}
+
+// Checks the privilege escalation of child
+// When commands try to make use of features that are unsupported by child context, they fail.
+func TestPrivilegesEscalation(d client.TestDPEInstance, c client.DPEClient, t *testing.T) {
```
Comment (jhand2) [inline verification/testing/deriveContext.go:162]: Same, I think this leaves dangling contexts.

File: verification/testing/deriveContext.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+	// by child context, it will fail.
+	if _, err = c.CertifyKey(handle, make([]byte, digestLen), client.CertifyKeyX509, client.CertifyAddIsCA); err == nil {
+		t.Errorf("[ERROR]: Should return %q, but returned no error", client.StatusInvalidArgument)
+	} else if !errors.Is(err, client.StatusInvalidArgument) {
+		t.Errorf("[ERROR]: Incorrect error type. Should return %q, but returned %q", client.StatusInvalidArgument, err)
+	}
+}
+
+// Checks whether the number of derived contexts (TCI nodes) are limited by MAX_TCI_NODES attribute of the profile
+func TestMaxTCIs(d client.TestDPEInstance, c client.DPEClient, t *testing.T) {
```
Comment (jhand2) [inline verification/testing/deriveContext.go:206]: Same. Irreversible.

File: verification/client/abi.go [hunk 4]
```diff
@@ -10,6 +10,9 @@ import (
 // DefaultContextHandle is the default DPE context handle
 var DefaultContextHandle = ContextHandle{0}
 
+// RetiredContextHandle is the retired DPE context handle
+var RetiredContextHandle = ContextHandle{255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255}
```
[the remark this answers is not in the document]
Comment (sree-revoori1) [reply verification/client/abi.go:14]: Went with option 2.

File: verification/testing/deriveContext.go [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+	if resp, err = c.DeriveContext(handle, make([]byte, digestLen), client.MakeDefault, 0, 0); err != nil {
+		t.Errorf("[ERROR]: Error while creating child context handle: %s", err)
+	}
+	handle = &resp.NewContextHandle
+	if resp.NewContextHandle != client.DefaultContextHandle {
+		t.Fatalf("[FATAL]: Incorrect handle. Should return %v, but returned %v", client.DefaultContextHandle, *handle)
+	}
+
+	// When there is already a default handle, setting MakeDefault and RetainParentContext will cause error
+	// because there cannot be two default handles in a locality
```
[the remark this answers is not in the document]
Comment (sree-revoori1) [reply verification/testing/deriveContext.go:39]: Done

File: verification/testing/deriveContext.go (hunk 3 above)
Comment (sree-revoori1) [reply verification/testing/deriveContext.go:206]: Done

File: verification/testing/deriveContext.go (hunk 1 above)
Comment (sree-revoori1) [reply verification/testing/deriveContext.go:80]: Done, opened https://github.com/chipsalliance/caliptra-dpe/issues/308

File: verification/testing/verification.go [hunk 6]
```diff
@@ -131,6 +161,11 @@ var AllTestCases = []TestCase{
 	WrongLocalityTestCase,
 }
 
+var IrreversibleTestCases = []TestCase{
+	DeriveContextTestCase,
+	DeriveContextSimulationTestCase,
```
[the remark this answers is not in the document]
Comment (sree-revoori1) [reply verification/testing/verification.go:166]: Done

**Review by jhand2 — APPROVED** [review]

**Pull request merged**